hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMRecoverLeaseRequest.java

```diff
@@ -269,7 +269,9 @@ private RecoverLeaseResponse doWork(OzoneManager ozoneManager,
     // refresh last block pipeline
     ContainerWithPipeline containerWithPipeline =
         ozoneManager.getScmClient().getContainerClient().getContainerWithPipeline(finalBlock.getContainerID());
-    finalBlock.setPipeline(containerWithPipeline.getPipeline());
+    if (containerWithPipeline != null) {
```

Comment:
When will containerWithPipeline be null？ We should no allow this to happen if the container is a valid container.

Reply:
If [ignorePipeline](https://github.com/apache/ozone/blob/37213b7085be08721934161d8102f09f7671d0b4/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java#L72) is true, `keyTable/openKeyTable` doesn't store pipeline info. And hence after fetching from db it returns null. Looks like this will impact leaseRecovery if pipeline info is not stored.

Reply:
I see we are getting pipeline info from SCM in LeaseRecover.

Reply:
No, pipeline is get from block info returned by OM.

Reply:
Mocked pipeline info